### PR TITLE
Rename/Merge branch

### PR DESCRIPTION
**Pull request by thibaultduponchelle**
Add `merge_branch` and `rename_branch` methods.

Replacing #219

### CODE REVIEW

File: examples/rename_branch.pl [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+use Pithub::Repos;
```
Comment (oalders) [inline examples/rename_branch.pl:5]: `use Pithub::Repos ();`

File: lib/Pithub/Repos.pm [hunk 2]
```diff
@@ -85,6 +85,79 @@ sub branches {
...
+
+=back
+
+=cut
+
+sub rename_branch {
+    my ( $self, %args ) = @_;
+    croak 'Missing parameters: data' unless $args{data};
+    croak 'Missing parameters: branch' unless $args{branch};
+    croak 'Missing key in parameters: new_name' unless $args{data}->{new_name};
```
Comment (oalders) [inline lib/Pithub/Repos.pm:116]: Could you double check the `croak` messages here to make sure they're in line with the rest of the API?

File: lib/Pithub/Repos.pm [hunk 3]
```diff
@@ -85,6 +85,79 @@ sub branches {
...
+
+=back
+
+=cut
+
+sub merge_branch {
+    my ( $self, %args ) = @_;
+    croak 'Missing parameters: data' unless $args{data};
+    croak 'Missing key in parameters: base' unless $args{data}->{base};
+    croak 'Missing key in parameters: head' unless $args{data}->{head};
```
Comment (oalders) [inline lib/Pithub/Repos.pm:152]: Could you double check the `croak` messages here to make sure they're in line with the rest of the API?

File: examples/merge_branch.pl [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+use Pithub::Repos;
```
[the remark this answers is not in the document]
Comment (thibaultduponchelle) [reply examples/merge_branch.pl]: Fixed

File: examples/rename_branch.pl (hunk 1 above)
Comment (thibaultduponchelle) [reply examples/rename_branch.pl]: Fixed

File: lib/Pithub/Repos.pm (hunk 3 above)
Comment (thibaultduponchelle) [reply lib/Pithub/Repos.pm]: I aligned with other :+1:

File: examples/list_branches.pl [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+use Pithub::Repos;
```
[the remark this answers is not in the document]
Comment (thibaultduponchelle) [reply examples/list_branches.pl:5]: Fixed

File: lib/Pithub/Repos.pm (hunk 2 above)
Comment (thibaultduponchelle) [reply lib/Pithub/Repos.pm]: I aligned with the rest of API :+1:

**Comment by thibaultduponchelle** [comment]
> I'm sorry for the delay in reviewing this.

You don't have to 😄  thank you for reviewing! 🙏🏼

**Review by oalders — APPROVED** [review]
⭐

**Pull request merged**